### ISSUE TITLE
Huge Traffic from Kubernetes masters to Spinnaker

### ISSUE THREAD

**Issue by anoopl** [spinnaker/spinnaker#5965]
### Issue Summary:
Huge Traffic to Kubernetes masters from Spinnaker
Spinnaker might be reading the status of deployments

### Cloud Provider(s):
Kubernetes/ GKE

### Environment:
Dev/test/prod

### Description:
Spinnaker receives a huge traffic from GKE masters, Spinnaker is on premise and connected through VPN.
There is around 2 TB data transfer daily. We tried to set Kind, but that did not make much difference. We noticed this after upgrade to Spinnker 1.12 version. We think it was normal on the 1.5 years older version we used.

### Steps to Reproduce:
Run spinnaker 1.12 with more than 50 namespaces and 5 or more deployments per namespace
This happens on all our environments
### Additional Details:

Our provider config:

  - name: dev
        requiredGroupMembership: []
        providerVersion: V2
        permissions: {}
        dockerRegistries: []
        configureImagePullSecrets: true
        serviceAccount: false
        cacheThreads: 1
        namespaces: []
        omitNamespaces: []
        kinds:
        - Deployment
        - ConfigMap
        - Secret
        - Service
        - VirtualService
        - DestinationRule
        - ServiceAccount
        omitKinds: []
        customResources: []
        cachingPolicies: []
        kubeconfigFile: /home/******/dev/kubeconfiig
        checkPermissionsOnStartup: true
         liveManifestCalls: true
        oAuthScopes: []
        onlySpinnakerManaged: true
---

**Comment by ezimanyi** [comment]
@anoopl : In general we're aware that Spinnaker transfers a lot of data from the cluster; 2 TB per day seems like a lot, though without full context on how many resources Spinnaker is caching it's hard to know for sure if that's reasonable.

Some things you can do to reduce the amount of data are:
* Reduce the frequency that your cluster is being cache, by setting `cacheIntervalSeconds` to something larger than the default 60s on the affected account(s).  (This will affect the display of resources in the UI which depends on the cache.)
* Reduce the namespaces or kinds that you are caching using `kinds`/`omitKinds` or `namespaces`/`omitNamespaces`.  If there are a lot of resources in a particular namespace, or of a particular kind, and you don't need to deploy to that namespace or deploy that kind, turning of caching can reduce the amount of data being transferred from your cluster.

Another possibility is if you have multiple accounts caching the same cluster/namespace.  Accounts in Spinnaker operate independently, so if you set up multiple accounts to point to the same cluster, they will repeat the work of caching that cluster.  (Some people do this but then restrict each account to a subset of namespaces, which of course doesn't duplicate work in that case.)

I am also working in #5905, which should greatly reduce the amount of data we need to cache from your cluster (because most of the time we don't even use the cache; the hit rate is extremely low).  That will probably have the most effect, but until then hopefully some of the ideas above will help.

**Comment by anoopl** [comment]
@ezimanyi  Thank you so much for the reply. 

We already using omit namespaces and kind to cache. Let us try setting the cacheIntervalSeconds to something like 3 to 5 minutes and see if it can reduce some data. We do not have multiple accounts. We have only one account per cluster.

How can we find more details about caching, like how many objects its caching and which objects?

**Comment by ezimanyi** [comment]
@anoopl : You might find some useful information in the `clouddriver` logs, as at the end of each caching agent's execution it prints something like:
```
k8s/KubernetesCoreCachingAgent[1/1]: grouping service has 8 entries and 71 relationships
k8s/KubernetesCoreCachingAgent[1/1]: grouping persistentVolumeClaim has 2 entries and 0 relationships
...
```
You could look to see whether there are particular kubernetes kinds that have a lot of entries and further investigate those.  Of course it's not just the number of entries, as `clouddriver` is fetching the full resource---so if there are some kinds that tend to have a lot of data (ex: `ConfigMap`s that are very large) then these would especially contribute to the amount of data transferred.

**Comment by ezimanyi** [comment]
I'm going to close this as the root cause will be fixed by #5905.  Of course if there are any workarounds/ideas in the meantime they can be posted here, but we'll leave #5905 as the open issue that will ultimately reduce the amount of data we fetch from clusters.
